rosbag2_transport/test/rosbag2_transport/test_play_until.cpp

```diff
@@ -154,6 +154,48 @@ TEST_F(RosBag2PlayUntilTestFixture, play_until_less_than_the_total_duration)
   EXPECT_EQ(replayed_topic1[1]->int32_value, 1);
 }
 
+TEST_F(RosBag2PlayUntilTestFixture, play_until_less_than_the_total_duration_message_order)
+{
+  auto primitive_message1 = get_messages_basic_types()[0];
+  auto primitive_message2 = get_messages_basic_types()[0];
+  primitive_message1->int32_value = 1;
+  primitive_message2->int32_value = 2;
+
+  auto topic_types = std::vector<rosbag2_storage::TopicMetadata>{
+    {1u, kTopic1Name_, "test_msgs/BasicTypes", "", {}, ""}};
+
+  std::vector<std::shared_ptr<rosbag2_storage::SerializedBagMessage>> messages =
+  {
+    serialize_test_message(kTopic1Name_, 10, 5, primitive_message1),
+    serialize_test_message(kTopic1Name_, 50, 45, primitive_message2),
```

Comment:
I would change the order of the timestamps to make sure that our online sorting is working as expected as well.
```suggestion
    serialize_test_message(kTopic1Name_, 10, 45, primitive_message1),
    serialize_test_message(kTopic1Name_, 50, 5, primitive_message2),
```

Reply:
I changed the values so that `send < recv` for the first message (even though it doesn't really change anything) while keeping the relative order you want, see d78f6f15fddad21d12b695591586c22fcf54e50c